src/org/thoughtcrime/securesms/ConversationActivity.java

```diff
@@ -1668,7 +1668,9 @@ public void initializeContactRequest() {
       messageRequestBottomView.setVisibility(View.GONE);
       composePanel.setVisibility(View.VISIBLE);
     });
+    DcEventCenter eventCenter = DcHelper.getEventCenter(this);
     messageRequestBottomView.setBlockOnClickListener(v -> {
+      eventCenter.removeObserver(DcContext.DC_EVENT_CONTACTS_CHANGED, this);
```

Comment:
even if it may be okay here, in general, i think, it is better not to keep unneeded strong references, so maybe better `DcHelper.getEventCenter(this).removeObserver(DcContext.DC_EVENT_CONTACTS_CHANGED, this);`.

i would also add a comment, sth. as "// avoid showing compose panel on receiving DC_EVENT_CONTACTS_CHANGED for the chat that is no longer a request after blocking"